File: PTC/ptc-main/ptc-main/lib/app/modules/supplier/supplier_model.dart
class Supplier {
  final Vehicle vehicle;
  final List<Passenger> passengers;

  Supplier({
    required this.vehicle,
    required this.passengers,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) => Supplier(
    vehicle: Vehicle.fromJson(json["vhcSpplier"]),
    passengers: List<Passenger>.from(json["psgSppliers"].map((x) => Passenger.fromJson(x))),
  );

  Map<String, dynamic> toJson() => {
    "vhcSpplier": vehicle.toJson(),
    "psgSppliers": List<dynamic>.from(passengers.map((x) => x.toJson())),
  };
}

class Vehicle {
  final int? id;
  final String vehicleId;
  final String vehicleType;
  final String company;
  final String vehicleImg;
  final int passengerCount;
  final DateTime inTime;
  final bool? flag;
  final DateTime? confirmOutTime;

  Vehicle({
    this.id,
    required this.vehicleImg,
    required this.vehicleId,
    required this.vehicleType,
    required this.company,
    required this.passengerCount,
    required this.inTime,
    this.flag,
    this.confirmOutTime,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) => Vehicle(
    id: json["id"],
    vehicleId: json["vehicleId"],
    vehicleType: json["vehicleType"],
    company: json["company"],
    vehicleImg: json["vehicleImg"],
    passengerCount: json["passangerCount"],
    inTime: DateTime.parse(json["inTime"]),
    flag: json["flag"],
    confirmOutTime: json["confirmOutTime"] == null ? null : DateTime.parse(json["confirmOutTime"]),
  );

  Map<String, dynamic> toJson() => {
    "vehicleId": vehicleId,
    "vehicleType": vehicleType,
    "company": company,
    "vehicleImg": vehicleImg,
    "passangerCount": passengerCount,
  };
}

class Passenger {
  final int? id;
  final int? vhcId;
  final String name;
  bool flag;

  Passenger({
    this.id,
    this.vhcId,
    required this.name,
    required this.flag,
  });

  factory Passenger.fromJson(Map<String, dynamic> json) => Passenger(
    id: json["id"],
    vhcId: json["vhcId"],
    name: json["name"],
    flag: json["flag"],
  );

  Map<String, dynamic> toJson() => {
    "name": name,
  };
}

File: PTC/PTC-master/PTC-master/PTC/Models/SupplierModel.cs
namespace PTC.Models
{
    public class PsgSpplier
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public int VhcId { get; set; }
        public bool Flag { get; set; }
    }

    public class VhcSpplier
    {
        public int Id { get; set; }
        public string? VehicleId { get; set; }
        public string? VehicleType { get; set; }
        public string? Company { get; set; }
        public int PassangerCount { get; set; }
        public DateTime InTime { get; set; } = DateTime.Now;
        public bool Flag { get; set; }
        public DateTime? ConfirmOutTime { get; set; } = null;
        public required byte[] VehicleImg { get; set; }
    }

    public class  SpplierTransactionRecord
    {
        public int Id { get; set; }
        public string Username { get; set; }
        public int LoginId { get; set; }
        public int TransactionId {  get; set; }
        public string Type { get; set; }
        public DateTime ScanTime { get; set; }
    }

    public class ConfirmVhc
    {
        public required int VhcId { get; set; }
        public required int[] PsgId { get; set; }
    }

    public class Spplier
    {
        public required VhcSpplier VhcSpplier { get; set; }
        public required PsgSpplier[] PsgSppliers { get; set; }
    }
}